packages/maskbook/src/plugins/RedPacket/SNSAdaptor/NftRedPacketHistoryItem.tsx

```diff
@@ -0,0 +1,186 @@
+        border: `solid 1px ${theme.palette.divider}`,
+        marginBottom: theme.spacing(1.5),
+        position: 'static !important' as any,
+        height: 'auto !important',
+        padding: theme.spacing(2),
+        backgroundColor: MaskColorVar.lightBackground,
+    },
+    primary: {
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'space-between',
+    },
+    secondary: {
+        fontSize: 12,
+    },
+    message: {
+        whiteSpace: 'nowrap',
+        overflow: 'hidden',
+        textOverflow: 'ellipsis',
+    },
+    strong: {
+        overflow: 'hidden',
+        textOverflow: 'ellipsis',
+    },
+    span: {
+        maxWidth: 350,
+        display: 'inline-flex',
+    },
+    time: {
+        fontSize: 12,
+        color: theme.palette.text.secondary,
+    },
+    box: {
+        display: 'flex',
+        width: '100%',
+    },
+    content: {
+        transform: 'translateY(-4px)',
+        width: '100%',
+        padding: theme.spacing(0, '1rem'),
+    },
+    section: {
+        display: 'flex',
+        width: '100%',
+        justifyContent: 'space-between',
+        alignItems: 'center',
+        marginBottom: theme.spacing(2),
+    },
+    icon: {
+        width: 27,
+        height: 27,
+    },
+    title: {
+        whiteSpace: 'break-spaces',
+        fontWeight: 500,
+        fontSize: 16,
+    },
+    info: {
+        fontSize: 14,
+        color: theme.palette.mode === 'light' ? '#5B7083' : '#c3cbd2',
+    },
+    nftList: {
+        width: 390,
+    },
+    actionButton: {
+        height: 26,
+        minHeight: 'auto',
+    },
+    footer: {
+        width: '100%',
+        display: 'flex',
+        flexWrap: 'wrap',
+        justifyContent: 'space-between',
+        marginTop: theme.spacing(2),
+    },
+    footerInfo: {
+        fontSize: 15,
+        color: theme.palette.mode === 'light' ? '#5B7083' : '#c3cbd2',
+        '& strong': {
+            color: theme.palette.text.primary,
+        },
+    },
+}))
+
+export interface NftRedPacketHistoryItemProps {
+    history: NftRedPacketHistory
+    onSend: (history: NftRedPacketHistory) => void
+}
+export const NftRedPacketHistoryItem: FC<NftRedPacketHistoryItemProps> = memo((props) => {
+    const account = useAccount()
+    const { history, onSend } = props
+    const { t } = useI18N()
+    const { classes } = useStyles()
+    const {
+        computed: { canSend, listOfStatus },
+    } = useNftAvailabilityComputed(account, history.payload)
+    const contractDetailed = useMemo(() => {
+        return { ...history.token.contractDetailed, address: history.token.address } as ERC721ContractDetailed
+    }, [history.token.contractDetailed, history.token.address])
+
+    const handleSend = useCallback(() => {
+        if (canSend) onSend(history)
+    }, [onSend, canSend, history])
+
+    const { value: redpacketStatus } = useAvailabilityNftRedPacket(history.rpid, account)
+    const bitStatusList = redpacketStatus
```

Comment:
I think it's better to pass the `tokenIds` into `useAvailabilityNftRedPacket` to calculate out `claimedList` directly.